interviewTest/Task1.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ This is simple list component with the following expected functionality:
+
+ 1. All items in the list are rendered correctly on the screen
+ 2. Pressing the button sorts the list and re-renders it
+ 3. The justifyContent prop is used to determine the Flex CSS layout
+
+ There are issues with this implementation. Can you see them?
+*/
+import React, { useState } from "react";
+import { Button, StyleSheet, Text, View } from "react-native";
+
+const data = ["Rick", "Morty", "Jerry", "Summer"];
+
+type Props = {
+  justifyContent: "flex-start" | "flex-end" | "center";
+};
+
+export const MyList = (props: Props) => {
+  const { justifyContent } = props;
+
+  const [listItems, setListItems] = useState(data);
+
+  const handleSort = () => setListItems(listItems.sort());
+
+  const flexStyle = { justifyContent };
+
+  return (
+    <View style={[flexStyle, styles.container]}>
+      {listItems.map((item: string, index: number) => (
+        <Text key={index}>{item + "is at index " + index}</Text>
```

Comment:
The issue here is that we're using indices for our `key` prop, so even if we re-arrange or shuffle `listItems`, the keys will always be `0, 1, 2, 3`, which will prevent React from realizing that there's a new diff and that it needs to update the DOM.

The optimal fix is to have some unique identifier for each element, with UUID4 being very popular for this use case. We don't have that in this particular data set, but I can notice that there are no repeating names in the list. Sof a fix here would be:
```suggestion
        <Text key={item}>{item + "is at index " + index}</Text>
```